test/unit/pcv/PCVDepositAggregator.test.ts

```diff
@@ -0,0 +1,444 @@
+      await token.mint(pcvDeposit3.address, ethers.utils.parseEther('1000'));
+
+      // Call deposit on each pcv deposit so that their balances update
+      await pcvDeposit1.deposit();
+      await pcvDeposit2.deposit();
+      await pcvDeposit3.deposit();
+
+      expect(await pcvDepositAggregator.getTotalBalance()).to.equal(ethers.utils.parseEther('10000'));
+      expect(await pcvDepositAggregator.totalWeight()).to.equal(100);
+
+      // Call rebalance
+      await pcvDepositAggregator.rebalance();
+
+      // Check pcv deposit balances
+      // Should be 2000, 3000, 4000 in deposits
+      // Should be 1000 in the aggregator
+      expect(await token.balanceOf(pcvDeposit1.address)).to.equal(ethers.utils.parseEther('2000'));
+      expect(await token.balanceOf(pcvDeposit2.address)).to.equal(ethers.utils.parseEther('3000'));
+      expect(await token.balanceOf(pcvDeposit3.address)).to.equal(ethers.utils.parseEther('4000'));
+
+      // Also check the aggregator balance
+      expect(await token.balanceOf(pcvDepositAggregator.address)).to.equal(ethers.utils.parseEther('1000'));
+    });
+
+    it('rebalances a single deposit', async () => {
+      // Mint 4000, 4000, and 4000 tokens to each pcv deposit, respectively
+      await token.mint(pcvDeposit1.address, ethers.utils.parseEther('6000'));
+      await token.mint(pcvDeposit2.address, ethers.utils.parseEther('3000'));
+      await token.mint(pcvDeposit3.address, ethers.utils.parseEther('1000'));
+
+      // Call deposit on each pcv deposit so that their balances update
+      await pcvDeposit1.deposit();
+      await pcvDeposit2.deposit();
+      await pcvDeposit3.deposit();
+
+      expect(await pcvDepositAggregator.getTotalBalance()).to.equal(ethers.utils.parseEther('10000'));
+      expect(await pcvDepositAggregator.totalWeight()).to.equal(100);
+
+      await pcvDepositAggregator.rebalanceSingle(pcvDeposit1.address);
+
+      expect(await token.balanceOf(pcvDeposit1.address)).to.equal(ethers.utils.parseEther('2000'));
+      expect(await token.balanceOf(pcvDepositAggregator.address)).to.equal(ethers.utils.parseEther('4000'));
+    });
+
+    it('adds a pcv deposit', async () => {
+      const mockPCVDepositDeployer = new MockPCVDepositV2__factory(impersonatedSigners[userAddress]);
+
+      const pcvDeposit4 = await mockPCVDepositDeployer.deploy(core.address, token.address, 0, 0);
+      await pcvDeposit4.deployTransaction.wait();
+
+      await pcvDepositAggregator.connect(impersonatedSigners[governorAddress]).addPCVDeposit(pcvDeposit4.address, 10);
+      expect(await pcvDepositAggregator.totalWeight()).to.equal(110);
+    });
+
+    it('removes a pcv deposit', async () => {
+      await pcvDepositAggregator.connect(impersonatedSigners[governorAddress]).removePCVDeposit(pcvDeposit1.address);
+      expect(await pcvDepositAggregator.totalWeight()).to.equal(80);
+    });
+
+    it('reports accurate targetPercentHeld', async () => {
```

Comment:
This test is also testing rebalancing. Would be useful to see this without a rebalance